<launch>
    <param name="use_sim_time" value="false" />

    <!-- Main node -->
    <node name="orb_slam3_stereo_inertial" pkg="orb_slam3_ros_wrapper" type="orb_slam3_ros_wrapper_stereo_inertial" output="screen">
        <!-- change the topics according to the dataset -->
        <remap from="/camera/left/image_raw"    to="/device_0/sensor_0/Infrared_1/image/data"/>
        <remap from="/camera/right/image_raw"   to="/device_0/sensor_0/Infrared_2/image/data"/>
        <remap from="/imu"                      to="/device_0/sensor_2/imu"/>

        <!-- Parameters for original ORB-SLAM3 -->
        <param name="voc_file"      type="string"   value="/home/hunter/Documents/ORB/ORB_SLAM3/Vocabulary/ORBvoc.txt" />
        <param name="settings_file" type="string" value="$(find orb_slam3_ros_wrapper)/config/RealSense_D455_stereo.yaml"/>

        <!-- Parameters for ROS -->
        <param name="world_frame_id"    type="string"   value="world" />
        <param name="cam_frame_id"      type="string"   value="camera" />
        <param name="imu_frame_id"      type="string"   value="imu" />
        <param name="enable_pangolin"   type="bool"     value="true" />
    </node>
    
    <!-- Visualization -->
    <!--<node name="rviz" pkg="rviz" type="rviz" args="-d $(find orb_slam3_ros_wrapper)/config/orb_slam3_with_imu.rviz" output="screen" />-->
    <!--
    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_orb_slam3" output="screen" ns="orb_slam3_ros_wrapper" >
        <param name="/target_frame_name" value="/world" />
        <param name="/source_frame_name" value="/imu" />
        <param name="/trajectory_update_rate" value="20.0" />
        <param name="/trajectory_publish_rate" value="20.0" />
    </node>
-->
    
</launch>